### ISSUE TITLE
Upgrade to Dokka 0.10.0

### ISSUE THREAD

**Issue by zach-klippenstein** [square/workflow#651]
Dokka 0.10.0 was just [released](https://github.com/Kotlin/dokka/releases/tag/0.10.0), fixes a bunch of bugs and introduces multiplatform support.

**Comment by zach-klippenstein** [comment]
This is currently blocked by https://github.com/vanniktech/gradle-maven-publish-plugin/issues/69 which tries to automatically apply the `dokka-android` plugin to `kotlin-android` modules.

Error message is:
```
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':workflow-ui-android'.
> Failed to apply plugin [id 'org.jetbrains.dokka-android']
   > Cannot add task 'dokka' as a task with that name already exists.
```

**Comment by zach-klippenstein** [comment]
When the maven publish plugin version 0.9.0 is published, we can re-open #652 to close this.

**Comment by zach-klippenstein** [comment]
According to https://github.com/vanniktech/gradle-maven-publish-plugin/pull/70, the 0.9.0 release of the maven publish plugin is blocked by https://github.com/vanniktech/gradle-maven-publish-plugin/issues/54.
